src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -279,13 +281,50 @@ internal void PublishRpcChannelReadyEvent(RpcEvent initEvent)
                 return;
             }
 
-            RpcChannelReadyEvent readyEvent = new RpcChannelReadyEvent(_workerId, _workerConfig.Language, this, _initMessage.WorkerVersion, _initMessage.Capabilities);
-            _eventManager.Publish(readyEvent);
+            var managedDepedencyProvider = new ManagedDependencyProvider();
+            var osPlatform = OSPlatform.Windows;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                osPlatform = OSPlatform.Linux;
+            }
 
-            if (_functionRegistrations != null)
+            var managedDependencies = managedDepedencyProvider.GetDependencies(Path.Combine(_rootScriptPath, ScriptConstants.HostMetadataFileName), osPlatform, new ManagedDependencyConfigs()).GetAwaiter().GetResult();
+            if (_initMessage.Capabilities.ContainsKey("ManagedDependenciesEnabled")
```

Comment:
Use `_initMessage.Capabilities.TryGetValue("ManagedDependenciesEnabled", out string value)` and evaluate the result instead

nit: replace the two last checks with `managedDependencies?.Any()`

Reply:
However I changed the last 2 condition to managedDependencies?.Any() == true